cli.md

```diff
@@ -36,7 +36,11 @@ The `executable` could involve multiple command line tokens (e.g., `npm run` or
 - You cannot legally specify any argument more than one time
 - All codemod rules are loaded by default unless `--codemode-include` specifies a list.  `-codemode-exclude` works off all default codemods.
 - Specifying a `--codemod-include`or `--codemod-exclude`  that references a non-existent codemod will result in an error
+- The `--path-include` and `--path-exclude` patterns are interpreted as relative to the given `<project directory>`. In practice this means that the patterns should be joined with the `<project directory>` when used internally and also when passed to external tools.
+- Codemodders are free to determine reasonable defaults for included and excluded paths. In general, codemodders should attempt to match only relevant source files and to ignore test directories and build artifacts by default. For example, codemodders will generally want to exclude `**/tests/**` by default. This will be interpreted relative to the given `<project directory>`, which means that the effective pattern will be `<project directory>/**/tests/**`.
 - For `--path-include` and `--path-exclude`, specific line numbers can be supplied. For instance, to include `src/Foo.java` but only allow changes found on line 11, you would pass `--path-include src/Foo.java:11`.
+- *Included* patterns that contain line numbers should be stripped of the line number before being used by either codemodder or external tools to determine which paths are included.
+- *Excluded* patterns that contain line numbers should *not* be used by either codemodder or external tools when determining paths to be excluded. In other words, a single excluded line should not prevent the entire file from being excluded by either codemodder or any external tools.
 - The line includes/excludes only only specifies if nodes are scanned/considered by the codemods. I won’t guarantee that nodes that matches the rules remains unchanges. For example, for `--path-exclude src/[Foo.java](http://Foo.java):11` any vulnerable node inside line `11` in `Foo.java` will be ignore by individual codemods. However it may be changed as part of a fix for another vulnerable node.
```

Comment:
sorry the wording isn't quite correct, specifically `being used by either codemodder or external tools` and `used by either codemodder or external tools`. It's **only for external tools**. Codemodder should use line numbers for both (otherwise we wouldn't have it in the cli at all). 

Basically any external tool (semgrep for now) should:
1. receive include paths without line numbers (give semgrep path without line numbers)
2. not receive exclude patterns with line numbers at all, neither the path nor the line number.

Reply:
I can try to clarify the wording here but it remains true that the line number needs to be removed when determining which _paths_ to include. The line number is only used by the codemod itself when already evaluating a selected file.